### PR TITLE
Add common tile component

### PR DESCRIPTION
**Pull request by SwiftyAlex**
Used throughout the majority of views, this one is pretty useful. Supports light/dark mode, VoiceOver, and has a convenience wrapper to handle buttons. Initialisers are a little funky but thats the cleanest way I've found thus far to have default args and a generic to support background styles.

Refactors colors so gradients wont need to be duplicated if we're directly accessing them via the fancy initialisers.
Uses my old friend stack switching to ensure good experiences for all text sizes.

Note that TalkCell has been left alone. These components are generally for the About, talker and contributor pages.

| 🌝 | 🌚 | 🔎 |
| - | - | - | 
|Screenshot 2022-07-05 at 21 13 11|Screenshot 2022-07-05 at 21 13 36|Screenshot 2022-07-05 at 21 23 06|

Note that big text here is in a scroll view.

| 🌝 | 🌚 | 🔎 |
| - | - | - | 
|Screenshot 2022-07-05 at 22 07 37|Screenshot 2022-07-05 at 22 08 05|Screenshot 2022-07-05 at 22 08 51|

### CODE REVIEW

File: SwiftLeeds/Views/Components/CommonTileView.swift [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            }
+        })
+        .padding(Padding.cell)
+        .frame(minHeight: Constants.compactCellMinimumHeight)
+        .background(
+            backgroundStyle,
+            in: RoundedRectangle(cornerRadius: Constants.cellRadius)
+        )
+        .accessibilityElement(children: .ignore)
+        .accessibilityLabel(
```
Comment (SwiftyAlex) [inline SwiftLeeds/Views/Components/CommonTileView.swift:73]: There's arguments for `
.accessibilityElement(children: .combine)` here but I prefer the manual control.

File: SwiftLeeds/Extension/SquishyButtonStyle.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//  SquishyButtonStyle.swift
+//  SwiftLeeds
+//
+//  Created by Alex Logan on 01/07/2022.
+//
+
+import Foundation
+import SwiftUI
+
+ struct SquishyButtonStyle: ButtonStyle {
```
Comment (pdamonkey) [inline SwiftLeeds/Extension/SquishyButtonStyle.swift:11]: Not an extension but a style, so should be put into its own group

File: SwiftLeeds/Views/Components/CommonTileButtton.swift [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+//  CommonTileButtton.swift
+//  SwiftLeeds
+//
+//  Created by Alex Logan on 05/07/2022.
+//
+
+import SwiftUI
+
+struct CommonTileButtton<BackgroundType: ShapeStyle>: View {
+    let primaryText: String
```
Comment (pdamonkey) [inline SwiftLeeds/Views/Components/CommonTileButtton.swift:11]: Is there another initialiser or should these be made private?

File: SwiftLeeds/Views/Components/CommonTileButtton.swift [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        Button(action: onTap) {
+            CommonTileView(
+                primaryText: primaryText,
+                secondaryText: secondaryText,
+                primaryColor: primaryColor,
+                secondaryColor: secondaryColor,
+                backgroundStyle: backgroundStyle
+            )
+        }
+        .buttonStyle(SquishyButtonStyle())
```
Comment (pdamonkey) [inline SwiftLeeds/Views/Components/CommonTileButtton.swift:61]: How is the onTap handled via VoiceOver?

**Review by pdamonkey — CHANGES_REQUESTED** [review]

File: SwiftLeeds/Views/Components/CommonTileButtton.swift (hunk 4 above)
Comment (SwiftyAlex) [reply SwiftLeeds/Views/Components/CommonTileButtton.swift:61]: Buttons work with voiceover for free, so onTap would be fired as expected.

File: SwiftLeeds/Views/Components/CommonTileView.swift [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+//  Created by Alex Logan on 05/07/2022.
+//
+
+import SwiftUI
+
+/// Generic primary secondary view
+struct CommonTileView<BackgroundType: ShapeStyle>: View {
+    @Environment(\.sizeCategory) var sizeCategory
+
+    let primaryText: String
```
[the remark this answers is not in the document]
Comment (SwiftyAlex) [reply SwiftLeeds/Views/Components/CommonTileView.swift:14]: The other initialisers are just conveniences, and making the above all private removes the ability to use the initialiser you get for free with a struct, so I avoid doing so.

File: SwiftLeeds/Views/Components/CommonTileButtton.swift (hunk 3 above)
Comment (SwiftyAlex) [reply SwiftLeeds/Views/Components/CommonTileButtton.swift:11]: The other initialisers are just conveniences, and making the above all private removes the ability to use the initialiser you get for free with a struct, so I avoid doing so.

File: SwiftLeeds/Views/Components/CommonTileButtton.swift (hunk 4 above)
Comment (pdamonkey) [reply SwiftLeeds/Views/Components/CommonTileButtton.swift:61]: Normally I'd use a hint to say what tapping the button does, but didn't realise it adopted the trait for free, nice

**Review by pdamonkey — APPROVED** [review]

File: SwiftLeeds/Views/Components/CommonTileButtton.swift (hunk 4 above)
Comment (SwiftyAlex) [reply SwiftLeeds/Views/Components/CommonTileButtton.swift:61]: Its a little scary to trust it at first, but the _generally_ automatic traits are a blessing

**Pull request merged**